app/controllers/setup_controller.rb

```diff
@@ -93,4 +94,15 @@ def minion_not_found(exception)
       format.json { render json: exception.message, status: :not_found }
     end
   end
+
+  def check_empty_settings
+    return unless settings_params.values.any?(&:empty?)
```

Comment:
No need to make the ActiveRecord models (so storage isn't relevant). Form objects can be plain old simple Ruby classes that validate the various fields. I think there are some good gems out there to help. Anyway, lets see this at a later time if we get more of this.